CRM/Custom/Form/Field.php

```diff
@@ -322,7 +322,7 @@ public function buildQuickForm() {
       'return' => ['title'],
     ];
 
-    if ($this->_action == CRM_Core_Action::UPDATE) {
+    if ($this->_action == CRM_Core_Action::UPDATE && !empty($this->_values['option_group_id'])) {
```

Comment:
Think this should actually be:
```
    if ($this->_action == CRM_Core_Action::UPDATE) {
      $this->freeze('data_type');
      if (!empty($this->_values['option_group_id'])) {
        // Before dev/core#155 we didn't set the is_reserved flag properly, which should be handled by the upgrade script...
        //  but it is still possible that existing installs may have optiongroups linked to custom fields that are marked reserved.
        $optionGroupParams['id'] = $this->_values['option_group_id'];
        $optionGroupParams['options']['or'] = [["is_reserved", "id"]];
      }
    }
```
Otherwise the datatype will not be frozen on update for custom fields without optiongroup